src/bitset.hpp

```diff
@@ -51,11 +57,14 @@ class Bitset {
   void operator&=(const Bitset &other);
   void operator|=(const Bitset &other);
 
+  friend std::ostream &operator<<(std::ostream &os, const Bitset &bitset);
```

Comment:
I did a search and couldn't the other examples in the codebase.
In this instance, we are overloading a operator definition for std::ostream, not Bitset.   We could define this outside the class definition, rather than as a friend member of Bitset.  However, this would be treated as an external function and we couldn't access private members of Bitset.  I could do that as I'm not accessing private members of Bitset, this is just the way I've seen it done.

https://docs.microsoft.com/en-us/cpp/standard-library/overloading-the-output-operator-for-your-own-classes?view=msvc-160

Reply:
In `gp_operation.hpp`, for example. I'm happy to do what seems best though I like consistency. We can make an entry in the contributing document.

Reply:
I think it would make more sense to make all ostream functions friends.  It gives us the option to access to private members if we ever need to.  Plus, while not technically a member of the class, I feel like it makes more sense when it is found under the class definition.